applications/plugins/image/image_gui/CMakeLists.txt

```diff
@@ -23,13 +23,18 @@ set(SOURCE_FILES
 include(../imagetoolbox/imagetoolbox.cmake)
 
 find_package(SofaGuiQt REQUIRED QUIET)
-find_package(Qt5 COMPONENTS Core Gui Widgets QUIET)
-
-if(Qt5Core_FOUND)
-    qt5_wrap_cpp(MOC_FILES ${MOC_HEADER_FILES})
+find_package(Qt5 COMPONENTS Core QUIET)
+if (NOT Qt5Core_FOUND)
+    if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.16")
+        find_package(Qt6 COMPONENTS Core CoreTools REQUIRED QUIET)
+        if (Qt6Core_FOUND)
+            qt6_wrap_cpp(MOC_FILES ${MOC_HEADER_FILES})
+            message(FATAL_ERROR "Image_gui is not compatible with Qt6 yet.") # to remove once image_gui has been updated
+        endif()
+        # else() should never happened as it would already break in SofaGuiQt
+    endif()
```

Comment:
```suggestion
    else()
        message(FATAL_ERROR "Compatibility with Qt version <5 is not supported anymore")
    endif()
```

Reply:
even if not possible according to your comment.

Reply:
The else() in this case means that Qt5 and Qt6 have not been found, so I dont think the new message is relevant ...

Reply:
If I am not wrong, this `endif` corresponds to the check of the `CMAKE_VERSION`

Reply:
well my comment was for the above endif()
but you are right I should maybe add a else() for cmake < 3.16

Reply:
well my remark was more on what happen if you build with qt4. As you removed the ```else qt4_wrap_cpp(...```
and you are looking for if (NOT Qt5Core_FOUND)  otherwise if (NOT Qt6Core_FOUND)